packages/client/lib/sentinel/index.ts

```diff
@@ -0,0 +1,1495 @@
+      if (!this.emit('topology-change', event)) {
+        this.self.#trace(`RedisSentinel: re-emit for topology-change for ${event.type} event returned false`);
+      }
+    });
+  }
+
+  static factory<
+    M extends RedisModules = {},
+    F extends RedisFunctions = {},
+    S extends RedisScripts = {},
+    RESP extends RespVersions = 2,
+    TYPE_MAPPING extends TypeMapping = {}
+  >(config?: SentinelCommander<M, F, S, RESP, TYPE_MAPPING>) {
+    const Sentinel = attachConfig({
+      BaseClass: RedisSentinel,
+      commands: COMMANDS,
+      createCommand: createCommand<ProxySentinel>,
+      createModuleCommand: createModuleCommand<NamespaceProxySentinel>,
+      createFunctionCommand: createFunctionCommand<NamespaceProxySentinel>,
+      createScriptCommand: createScriptCommand<ProxySentinel>,
+      config
+    });
+
+    Sentinel.prototype.Multi = RedisSentinelMultiCommand.extend(config);
+
+    return (options?: Omit<RedisSentinelOptions, keyof Exclude<typeof config, undefined>>) => {
+      // returning a "proxy" to prevent the namespaces.self to leak between "proxies"
+      return Object.create(new Sentinel(options)) as RedisSentinelType<M, F, S, RESP, TYPE_MAPPING>;
+    };
+  }
+
+  static create<
+    M extends RedisModules = {},
+    F extends RedisFunctions = {},
+    S extends RedisScripts = {},
+    RESP extends RespVersions = 2,
+    TYPE_MAPPING extends TypeMapping = {}
+  >(options?: RedisSentinelOptions<M, F, S, RESP, TYPE_MAPPING>) {
+    return RedisSentinel.factory(options)(options);
+  }
+
+  withCommandOptions<
+    OPTIONS extends CommandOptions<TYPE_MAPPING>,
+    TYPE_MAPPING extends TypeMapping,
+  >(options: OPTIONS) {
+    const proxy = Object.create(this);
+    proxy._commandOptions = options;
+    return proxy as RedisSentinelType<
+      M,
+      F,
+      S,
+      RESP,
+      TYPE_MAPPING extends TypeMapping ? TYPE_MAPPING : {}
+    >;
+  }
+
+  private _commandOptionsProxy<
+    K extends keyof CommandOptions,
+    V extends CommandOptions[K]
+  >(
+    key: K,
+    value: V
+  ) {
+    const proxy = Object.create(this.self);
+    proxy._commandOptions = Object.create(this.self.#commandOptions ?? null);
+    proxy._commandOptions[key] = value;
+    return proxy as RedisSentinelType<
+      M,
+      F,
+      S,
+      RESP,
+      K extends 'typeMapping' ? V extends TypeMapping ? V : {} : TYPE_MAPPING
+    >;
+  }
+
+  /**
+   * Override the `typeMapping` command option
+   */
+  withTypeMapping<TYPE_MAPPING extends TypeMapping>(typeMapping: TYPE_MAPPING) {
+    return this._commandOptionsProxy('typeMapping', typeMapping);
+  }
+
+  connect() {
+    return this.self.#internal.connect();
+  }
+
+  async _execute<T>(
+    execType: execType | undefined,
+    isReadonly: boolean | undefined,
+    fn: (client: RedisClient<RedisModules, RedisFunctions, RedisScripts, RespVersions, TypeMapping>) => Promise<T>
+  ): Promise<T> {
+    let clientInfo: clientInfo | undefined;
```

Comment:
Share the master lease between coucurrent runs

Reply:
implemented this and pushed, but not sure it makes sense as is.

we only get a lease on a write command so a pipeline of read/write/read will only get the leased client on the second command.  I think it might be better the way it was before.